src/general.py

```diff
@@ -26,3 +30,130 @@ def get_resources() -> dict:
             unique_resources[resource["GLIDEIN_ResourceName"]] = 1
 
     return unique_resources
+
+def make_working_subdirs(tests_dir: Path, working_dir: Path, curr_time: str, run=False):
+    """
+    Usage: create working subdirs in working_dir for each dir in tests_dir
+    @param tests_dir: directory containing all exerciser tests
+    @param working_dir: directory for storing info on exerciser runs
+    @param curr_time: time in the format: %Y-%m-%d_%H:%M
+    @param run: whether or not to submit the test jobs after creating working subdirs
+    """
+
+    # create top level working dir for exerciser run
+    timestamp_subdir = os.path.join(working_dir, curr_time)
+    if not os.path.exists(timestamp_subdir):
+        os.makedirs(timestamp_subdir)
+    else:
+        # need to wait 1 min to allow for unique dir names
+            print("Error: Please wait at least 1 minute between succesive runs")
+            print("Exiting...")
+            sys.exit(1)
+
+    # create subdir for each test in tests_dir
+    for test in tests_dir.iterdir():
+        # make working subdir for test
+        working_subdir = os.path.join(timestamp_subdir, test.name)
+        os.makedirs(working_subdir)
+
+        # copy contents of test dir into working subdir
+        sub_file = copy_test_dir(test, working_subdir)
+
+        # create text file with list of currently available ResourceNames
+        with open(os.path.join(working_subdir, "resource_list.txt"), "w") as resource_list:
+            resources = get_resources()
+            for resource in resources:
+                resource_list.write(f"{resource}\n")
```

Comment:
Building the resources list is costly (querying for all slot ads in the OSPool) so we do not want to do this for every test selected to run. This should be moved to before the `test_dir` iteration, and the file should be written once to `timestamp_subdir` not each `working_subdir`